nil pointer dereference panic

Version: 115268b

When I attempt to run sqlclosecheck  on github.com/go-sql-driver/mysql, I get the following crash:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1214f46]

goroutine 12 [running]:
github.com/ryanrolds/sqlclosecheck/pkg/analyzer.getAction(0x14b7620, 0xc001da7400, 0xc0020354b0, 0x2, 0x2, 0xc002038a60, 0xc001da1e10)
	[...]/sqlclosecheck/pkg/analyzer/analyzer.go:207 +0x4b6
github.com/ryanrolds/sqlclosecheck/pkg/analyzer.checkClosed(0xc001dba5c8, 0xc0020354b0, 0x2, 0x2, 0x2)
	[...]/sqlclosecheck/pkg/analyzer/analyzer.go:169 +0x88
github.com/ryanrolds/sqlclosecheck/pkg/analyzer.run(0xc001d4fb00, 0x7612e97, 0x17247c0, 0xc001fc2798, 0x0)
	[...]/sqlclosecheck/pkg/analyzer/analyzer.go:61 +0x216
golang.org/x/tools/go/analysis/unitchecker.run.func5.1()
	[...]/pkg/mod/golang.org/x/tools@v0.0.0-20200625211823-6506e20df31f/go/analysis/unitchecker/unitchecker.go:350 +0x6d8
sync.(*Once).doSlow(0xc001ce19f0, 0xc0004ef728)
	/usr/local/Cellar/go/1.14.5/libexec/src/sync/once.go:66 +0xec
sync.(*Once).Do(...)
	/usr/local/Cellar/go/1.14.5/libexec/src/sync/once.go:57
golang.org/x/tools/go/analysis/unitchecker.run.func5(0xc00011b450, 0x0)
	[...]/pkg/mod/golang.org/x/tools@v0.0.0-20200625211823-6506e20df31f/go/analysis/unitchecker/unitchecker.go:303 +0x1b3
golang.org/x/tools/go/analysis/unitchecker.run.func6.1(0xc0004f9958, 0xc000ebe6b0, 0xc00011b450)
	[...]/pkg/mod/golang.org/x/tools@v0.0.0-20200625211823-6506e20df31f/go/analysis/unitchecker/unitchecker.go:362 +0x33
created by golang.org/x/tools/go/analysis/unitchecker.run.func6
	[...]/pkg/mod/golang.org/x/tools@v0.0.0-20200625211823-6506e20df31f/go/analysis/unitchecker/unitchecker.go:361 +0xa3
```

--- a/pkg/analyzer/analyzer.go
+++ b/pkg/analyzer/analyzer.go
@@ -207,9 +207,12 @@ func getAction(instr ssa.Instruction, targetTypes []*types.Pointer) string {
 		}
 
 		isTarget := false
-		receiver := instr.Call.StaticCallee().Signature.Recv()
-		if receiver != nil {
-			isTarget = isTargetType(receiver.Type(), targetTypes)
+		staticCallee := instr.Call.StaticCallee()
+		if staticCallee != nil {
+			receiver := instr.Call.StaticCallee().Signature.Recv()
+			if receiver != nil {
+				isTarget = isTargetType(receiver.Type(), targetTypes)
+			}
 		}
 
 		name := instr.Call.Value.Name()

--- a/pkg/analyzer/analyzer.go
+++ b/pkg/analyzer/analyzer.go
@@ -33,7 +33,10 @@ func NewAnalyzer() *analysis.Analyzer {
 }
 
 func run(pass *analysis.Pass) (interface{}, error) {
-	pssa := pass.ResultOf[buildssa.Analyzer].(*buildssa.SSA)
+	pssa, ok := pass.ResultOf[buildssa.Analyzer].(*buildssa.SSA)
+	if !ok {
+		return nil, nil
+	}
 
 	// Build list of types we are looking for
 	targetTypes := getTargetTypes(pssa, sqlPackages)
@@ -228,10 +231,11 @@ func getAction(instr ssa.Instruction, targetTypes []*types.Pointer) string {
 
 		for _, aRef := range *instr.Addr.Referrers() {
 			if c, ok := aRef.(*ssa.MakeClosure); ok {
-				f := c.Fn.(*ssa.Function)
-				for _, b := range f.Blocks {
-					if checkClosed(&b.Instrs, targetTypes) {
-						return "handled"
+				if f, ok := c.Fn.(*ssa.Function); ok {
+					for _, b := range f.Blocks {
+						if checkClosed(&b.Instrs, targetTypes) {
+							return "handled"
+						}
 					}
 				}
 			}
@@ -280,10 +284,10 @@ func checkDeferred(pass *analysis.Pass, instrs *[]ssa.Instruction, targetTypes [
 
 			for _, aRef := range *instr.Addr.Referrers() {
 				if c, ok := aRef.(*ssa.MakeClosure); ok {
-					f := c.Fn.(*ssa.Function)
-
-					for _, b := range f.Blocks {
-						checkDeferred(pass, &b.Instrs, targetTypes, true)
+					if f, ok := c.Fn.(*ssa.Function); ok {
+						for _, b := range f.Blocks {
+							checkDeferred(pass, &b.Instrs, targetTypes, true)
+						}
 					}
 				}
 			}

